contracts/eosiolib/datastream.hpp

```diff
@@ -254,6 +254,24 @@ DataStream& operator >> ( DataStream& ds, std::array<T,N>& v ) {
    return ds;
 }
 
+template<typename DataStream, typename T, std::size_t N>
```

Comment:
I would prefer at least a specialization for `char (&v)[N]` to match the `vector<char>` specializations below.  However I think even those could be expanded  to be `is_scalar && !is_pointer` style overloads

Reply:
I just did the T (&v)[N} optimizations for is_scalar && !is_pointer, since the wasm compiler was exploding on the vector<T> optimization.

I also added an optimization for std::string.